Paymetheus/PurchaseTickets.xaml

```diff
@@ -0,0 +1,262 @@
+                    <TextBlock.Style>
+                        <Style TargetType="TextBlock">
+                            <Setter Property="Visibility" Value="Hidden" />
+                            <Setter Property="FontSize" Value="14"/>
+                            <Setter Property="Foreground" Value="#FF0C1E3E"/>
+                            <Setter Property="TextAlignment" Value="Left"/>
+                            <Setter Property="HorizontalAlignment" Value="Left"/>
+                            <Setter Property="TextWrapping" Value="Wrap"/>
+                            <Style.Triggers>
+                                <DataTrigger Binding="{Binding IsChecked, ElementName=UsingAPoolCheckBox}" Value="True">
+                                    <Setter Property="Visibility" Value="Visible" />
+                                </DataTrigger>                  
+                            </Style.Triggers>
+                        </Style>
+                    </TextBlock.Style>
+                </TextBlock>
+
+                <TextBox Grid.Column="3" Grid.Row="3" Name="PoolAddressTextBox" Height="20" VerticalAlignment="Center" TextAlignment="Left" Text="{Binding PoolAddressString, Mode=TwoWay, ValidatesOnExceptions=True}">
+                    <TextBox.Style>
+                        <Style TargetType="TextBox">
+                            <Setter Property="Visibility" Value="Hidden" />
+                            <Setter Property="Text" Value="" />
+                            <Style.Triggers>
+                                <!-- Zero this field if it becomes unchecked -->
+                                <DataTrigger Binding="{Binding IsChecked, ElementName=UsingAPoolCheckBox}" Value="False">
+                                    <Setter Property="Visibility" Value="Hidden" />
+                                    <Setter Property="Text" Value="" />
+                                </DataTrigger>
+                                <DataTrigger Binding="{Binding IsChecked, ElementName=UsingAPoolCheckBox}" Value="True">
+                                    <Setter Property="Visibility" Value="Visible" />
+                                </DataTrigger>
+                                <Trigger Property="Validation.HasError" Value="True">
+                                    <Setter Property="ToolTip" Value="{Binding RelativeSource={RelativeSource Self}, Path=(Validation.Errors)[0].ErrorContent}"/>
+                                </Trigger>
+                            </Style.Triggers>
+                        </Style>
+                    </TextBox.Style>
+                </TextBox>
+
+                <TextBlock Grid.Column="1" Grid.Row="4" Text="Pool fees" HorizontalAlignment="Right" VerticalAlignment="Center">
+                    <TextBlock.Style>
+                        <Style TargetType="TextBlock">
+                            <Setter Property="Visibility" Value="Hidden" />
```

Comment:
all of these setters should be in the TextBlock itself, instead of set using setters in the Style. Save the Style element for just the triggers.